oximeter/collector/src/results_sink.rs

```diff
@@ -98,6 +99,41 @@ pub async fn database_inserter(
                     );
                 }
             }
+
+            // Our internal testing rack will be running a ClickHouse cluster
+            // alongside a single-node installation for a while. We want to handle
+            // the case of these two installations running alongside each other, and
+            // oximeter writing to both of them. On our production racks ClickHouse
+            // will only be run on single-node modality, so we want to ignore all
+            // cases where the `ClickhouseClusterNative` service is not available.
+            if let Some(cluster_client) = &cluster_client {
+                match cluster_client.ping().await {
+                    Ok(()) => {
+                        debug!(
+                            log,
+                            "inserting {} samples into cluster database",
+                            batch.len();
+                        );
+                        match cluster_client.insert_samples(&batch).await {
+                            Ok(()) => trace!(
+                                log,
+                                "successfully inserted samples into cluster";
+                            ),
+                            Err(e) => {
+                                warn!(
+                                    log,
+                                    "failed to insert some results into metric cluster DB: {e}"
```

Comment:
Same thing here, we probably want `InlineErrorChain` on this. The stringification predates that type by a lot, but it'd be nice to get as much detail as we can now.